pkg/providers/openai/openaiclient.go

```diff
@@ -0,0 +1,80 @@
+// TODO: Explore resource pooling
+// TODO: Optimize Type use
+// TODO: Explore Hertz TLS & resource pooling
+// OpenAI package provide a set of functions to interact with the OpenAI API.
+package openai
+
+import (
+	"errors"
+	"fmt"
+
+	"glide/pkg/providers"
+
+	"github.com/go-playground/validator/v10"
```

Comment:
Hertz comes with its own validation mechanism, so I was thinking of trying to leverage it for the rest of validations we need to have in the system to keep one library per problem if that makes sense.

Reply:
This? https://github.com/cloudwego/hertz-examples/tree/main/binding

Reply:
Or/and: https://github.com/bytedance/go-tagexpr/tree/master/validator

Reply:
@mkrueger12 yes, I meant Hertz binder from https://github.com/cloudwego/hertz-examples/tree/main/binding . Would be cool if we used one validator uniformly across config/schema validations. Have not yet evaluated if that's feasible. We can table this discussion until we have clear understanding how to go there.